systems-management/nisysmgmt.yml

```diff
@@ -0,0 +1,2093 @@
+                - http://download.ni.com/ni-linux-rt/feeds/SystemLink/salt/18.5/x64/x64/           
+            osArchitectures:
+              type: array 
+              description: An array of strings representing OS architectures supported by a system(s) in descending priority.
+              items:
+                type: string
+              example:
+                - x64
+                - core2-64
+                - x86_64
+                - noarch
+                - any
+                - all
+        required: [feedUris]
+    required: [systemConfigurations]
+
+  PackageResponse:
+    title: Get Available Packages Response
+    description: A list with the available packages.
+    type: object
+    properties:
+      arch:
+        type: string
+        description: The architecture of the package.
+        example: windows_x64
+      breaks:
+        type: array
+        description: An array containing information about other packages this package breaks.
+        items:
+          type: string
+          description: The name of a package that this package breaks. This also contains information about the version of the package.
+          example: "ni-sysapi (>= 18.0.0)"
+      conflicts:
+        type: array
+        description: An array containing information about other packages this package conflicts with.
+        items:
+          type: string
+          description: The name of a package that this package conflicts with. This also contains information about the version of the package.
+          example: "ni-msvcrt-2015 (>= 14.1.4)"
+      depends:
+        type: array
+        description: An array containing information about other packages this package depends on.
+        items:
+          type: string
+          description: The name of a package that this package depends on. This also contains information about the version of the package.
+          example: "ni-metauninstaller (>= 18.0.1)"
+      description:
+        type: string
+        description: The description of the package.
+        example: This is an infrastructure package for NI Software.
+      displayName:
+        type: string
+        description: The display name of the package.
+        example: NI Security Update (KB 67L8LCQW)
+      displayVersion:
+        type: string
+        description: The display version of the package.
+        example: 2.0.0
+      enhances:
+        type: array
+        description: An array containing information about other packages this package enhances.
+        items:
+          type: string
+          description: The name of a package that this package enhances. This also contains information about the version of the package.
+          example: "ni-msiproperties (>= 18.0.1)"
+      essential:
+        type: integer
+        format: int32
+        description: Whether the package is essential or not. A value greater than 0 means the package is essential.
+        example: 1
```

Comment:
Is there a reason why this isn't a `boolean` instead of an `integer`? Based on the description, it sounds like a bool to me.

Reply:
NIPM C API gives us a numeric value. You can check it here: Perforce\AST\PackageManagement\components\nipkg\export\19.0\19.0.0f0\includes\nipkg_client\shared\constants.h 
The current implementation uses an integer and I thought I can leave it this way. But looking into C# API of NIPM I see that they made it a boolean. So after all it seems it should have been a boolean. I will change it.